UI/DesignWnd.cpp

```diff
@@ -3053,6 +3182,16 @@ void DesignWnd::MainPanel::DesignChanged() {
         m_confirm_button->Disable(true);
     }
     else {
+        m_replace_button->Disable(true);
+        if(m_replaced_design_id != ShipDesign::INVALID_DESIGN_ID) {
+            const ShipDesign& ship_design = *GetShipDesign(m_replaced_design_id);
+            std::string replace_name= ship_design.Name();
```

Comment:
ShipDesign::Name returns a const reference, I think, so you can store it as such instead of making a copy